argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,348 @@
+
+            summary[f"all:{str(abs_error_threshold)}"] = d1_all * 100
+            summary[f"fg:{str(abs_error_threshold)}"] = d1_fg * 100
+            summary[f"bg:{str(abs_error_threshold)}"] = d1_bg * 100
+            summary[f"all*:{str(abs_error_threshold)}"] = d1_all_est * 100
+            summary[f"fg*:{str(abs_error_threshold)}"] = d1_fg_est * 100
+            summary[f"bg*:{str(abs_error_threshold)}"] = d1_bg_est * 100
+
+        return summary
+
+
+def compute_disparity_error(
+    pred_fpath: Path,
+    gt_fpath: Path,
+    gt_obj_fpath: Path,
+    abs_error_thresholds: List = [10, 5, 3],
+    rel_error_thresholds: List = [0.1, 0.1, 0.1],
+    figs_fpath: Path = None,
+    save_disparity_error_image: bool = False,
+):
+    """Compute the disparity error metrics."""
+    pred_disparity = cv2.imread(str(pred_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+    gt_disparity = cv2.imread(str(gt_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+    gt_obj_disparity = cv2.imread(str(gt_obj_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+
+    pred_disparity = np.float32(pred_disparity) / 256.0
+    gt_disparity = np.float32(gt_disparity) / 256.0
+    gt_obj_disparity = np.float32(gt_obj_disparity) / 256.0
+
+    errors = accumulator(abs_error_thresholds)
+
+    # Compute masks
+    pred_mask = pred_disparity > 0
+    gt_mask = gt_disparity > 0
+    obj_mask = gt_obj_disparity > 0
+    bg_mask = np.logical_and(gt_mask, np.logical_not(obj_mask))
+    fg_mask = np.logical_and(gt_mask, obj_mask)
+
+    # If the density of the predicted disparity is less than 100%, then interpolate to fill up the holes.
+    num_pixels_all = pred_disparity.size
+    num_pixels_all_est = np.sum(pred_mask)
+    density = num_pixels_all_est / max(num_pixels_all, 1.0)
+
+    if density < 1.0:
+        pred_disparity = interpolate_disparity(pred_disparity)
+
+    # Compute errors
+    abs_err = np.abs(pred_disparity - gt_disparity)
+    rel_err = abs_err / np.maximum(gt_disparity, 1)
+
+    errors["num_pixels_bg"] = np.sum(bg_mask)
+    errors["num_pixels_fg"] = np.sum(fg_mask)
+    errors["num_pixels_bg_est"] = np.sum(bg_mask & pred_mask)
+    errors["num_pixels_fg_est"] = np.sum(fg_mask & pred_mask)
+
+    for abs_error_threshold, rel_error_threshold in zip(abs_error_thresholds, rel_error_thresholds):
+        bad_pixels = (abs_err > abs_error_threshold) & (rel_err > rel_error_threshold)
+
+        errors[f"num_errors_bg:{str(abs_error_threshold)}"] = np.sum(bg_mask & bad_pixels)
+        errors[f"num_errors_fg:{str(abs_error_threshold)}"] = np.sum(fg_mask & bad_pixels)
+        errors[f"num_errors_bg_est:{str(abs_error_threshold)}"] = np.sum(bg_mask & pred_mask & bad_pixels)
+        errors[f"num_errors_fg_est:{str(abs_error_threshold)}"] = np.sum(fg_mask & pred_mask & bad_pixels)
+
+    if save_disparity_error_image:
+        err = np.minimum(abs_err / abs_error_thresholds[0], rel_err / rel_error_thresholds[0])
```

Comment:
this function is getting quite long; should lines 233 - 281 go inside a helper function?

Reply:
Sure, the function is too long. I've broken it into `eval.py`, `utils.py`, and `constants.py`, similar to the Argoverse detection evaluation.